hyperlink/test/test_url.py

```diff
@@ -1100,3 +1100,37 @@ def test_from_text_bad_authority(self):
         self.assertRaises(URLParseError, URL.from_text, 'http://127.0.0.1:hi')
         # extra port colon (makes for an invalid host)
         self.assertRaises(URLParseError, URL.from_text, 'http://127.0.0.1::80')
+
+    def test_normalize(self):
+        url = URL.from_text('HTTP://Example.com/A%61/./../A%61?B%62=C%63#D%64')
+        assert url.get('Bb') == []
```

Comment:
This should be using `assertEquals`; the `assert` keyword will be discarded when run under `python -OO`, making it impossible to test this in a memory-constrained environment.